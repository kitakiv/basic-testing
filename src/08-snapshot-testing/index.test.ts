import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    const list = generateLinkedList([1]);
    const result = { value: 1, next: { value: null, next: null } };
    expect(list).toStrictEqual(result);
  });

  test('should generate linked list from values 2', () => {
    const list = generateLinkedList([1, 2]);
    expect(list).toMatchSnapshot({
      value: expect.anything(),
      next: expect.objectContaining({
        value: expect.anything(),
        next: expect.anything(),
      }),
    });
  });
});
