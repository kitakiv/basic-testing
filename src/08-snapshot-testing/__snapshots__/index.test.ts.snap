// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": ObjectContaining {
    "next": Anything,
    "value": Anything,
  },
  "value": Anything,
}
`;
